prov/util/src/util_shm.c

```diff
@@ -315,6 +322,8 @@ int smr_create(const struct fi_provider *prov, struct smr_map *map,
 
 void smr_free(struct smr_region *smr)
 {
+	if (smr->flags & SMR_FLAG_HMEM_ENABLED)
```

Comment:
Sometimes the check is against a flag; sometimes it's against caps.  It would be simpler to always use the same check, rather than requiring both options to be in sync and mean the same thing.